src/org/labkey/targetedms/outliers/OutlierGenerator.java

```diff
@@ -344,6 +374,24 @@ private void retainIfPreferred(Map<String, QCMetricConfiguration> preferredConfi
         return precursors;
     }
 
+    private boolean isExcludedPrecursorPeptide(Collection<ExcludedPrecursor> excludedPrecursors, String modifiedSeq, int charge, double mz)
+    {
+        return excludedPrecursors.stream().filter(ep -> Objects.equals(ep.getModifiedSequence(), modifiedSeq) &&
+                                                        Objects.equals(ep.getCharge(), charge) &&
+                                                        Objects.equals(ep.getMz(), mz)).count() == 1;
+    }
+
+    private boolean isExcludedPrecursorMolecule(Collection<ExcludedPrecursor> excludedPrecursors, String customIonName,
+                                                String ionFormula, double massMonoisotopic, double massAverage, int charge, double mz)
+    {
+        return excludedPrecursors.stream().filter(ep -> Objects.equals(ep.getCustomIonName(), customIonName) &&
+                                                        Objects.equals(ep.getIonFormula(), ionFormula) &&
+                                                        Objects.equals(ep.getMassMonoisotopic(), massMonoisotopic) &&
+                                                        Objects.equals(ep.getMassAverage(), massAverage) &&
+                                                        Objects.equals(ep.getCharge(), charge) &&
+                                                        Objects.equals(ep.getMz(), mz)).count() == 1;
```

Comment:
Here you're looking for exact matches in the numeric values but in the SQL you're rounding to 4 decimal places. Should we be consistent? Can we always use the exact match?

Reply:
Now using exact match, removed rounding to 4 decimal places. I was trying something out since i wasnt getting expected results, then forgot to remove the rounding.